### PR TITLE
DDA-7: Channels and data models

### PR DESCRIPTION
**Pull request by jvalanen**

### CODE REVIEW

File: docs/channels.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                    type: "object"
+                    schema:
+                      $ref: "#/definitions/diory"
+        error:
+          description: "Error response"
+          schema:
+            type: "string"
+            description: "Error message"
+definitions:
+  diory:
```
Comment (jvalanen) [inline docs/channels.yaml:31]: Should also room and home have own definitions?

File: docs/data-models.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+```
+id: uri
+rooms:
+    [path]: diory
+```
+Diory is in the room diograph with room id
+
+**Room**
+```
+id: uri
```
Comment (jvalanen) [inline docs/data-models.md:27]: Should this uri be room's path? The same as in home's room array?

File: electron/channels/get-room-channel.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const { ipcMain } = require('electron')
+const { channels } = require('../../src/shared/constants')
```
Comment (jvalanen) [inline electron/channels/get-room-channel.js:2]: I'm thinking if channelNames or channelIds is better... This sounds like it's the channel itself. But of course when you look at /shared/constants you understand that it's not... If you like it this way, it's ok for me.

File: electron/main.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 const { app, BrowserWindow, ipcMain } = require('electron');
 const path = require('path');
+const glob = require('glob');
 const url = require('url');
 const fs = require('fs');
 const { channels } = require('../src/shared/constants');
```
Comment (jvalanen) [inline electron/main.js]: There's now unnecessary imports here

**Comment by jvalanen** [comment]
Very important refactoring and documentation, to get good base!

Documentation can be pretty simple now. We can go into details later when things stabilize.

File: public/home.json [hunk 5]
```diff
@@ -2,12 +2,12 @@
...
   "rooms": {
-    "room1": {
+    "./public/room1": {
       "id": "room1",
       "text": "Room 1",
       "image": "https://pix6.agoda.net/hotelImages/4656079/-1/f7771c6afc7cc32401286116a7eed6f0.jpg?s=1024x768"
     },
-    "room2": {
+    "./public/room2": {
       "id": "room2",
```
Comment (jvalanen) [inline public/home.json]: So this is the id which identifies this room when it's shared to public/other people?

Should it be a bit more "id like", if not uuid, maybe something like: "room2-it2je7idkd3"

File: docs/channels.yaml (hunk 1 above)
Comment (opohjola) [reply docs/channels.yaml:31]: yes, if we think that this is is a good format. Is it?

File: docs/data-models.md (hunk 2 above)
Comment (opohjola) [reply docs/data-models.md:27]: it's room id, not path

File: electron/channels/get-room-channel.js (hunk 3 above)
Comment (opohjola) [reply electron/channels/get-room-channel.js:2]: this was used in template but I agree that they are misleading.

Comment (opohjola) [reply electron/channels/get-room-channel.js:2]: used also in ipcRenderer, so let's leave it

File: docs/channels.yaml (hunk 1 above)
Comment (jvalanen) [reply docs/channels.yaml:31]: I don't see any problem with this or "know about anything better" :)

File: docs/data-models.md (hunk 2 above)
Comment (jvalanen) [reply docs/data-models.md:27]: Yeah, I understand this better now.

But in Home definition there is still `[path]`? Is that still correct? Or should it be `[id]`?

File: docs/data-models.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+data: object
+links: 
+    [path]: diory
+```
+
+**Home**
+```
+id: uri
+rooms:
+    [path]: diory
```
Comment (jvalanen) [inline docs/data-models.md]: Should this be `[id]` if the connector is the one responsible of paths?

File: public/home.json (hunk 5 above)
Comment (opohjola) [reply public/home.json]: yes, it could be. Let's fix this in DDA-5

**Pull request merged**